src/l500/ac-trigger.cpp

```diff
@@ -531,17 +514,16 @@ namespace ivcam2 {
                 return;  // error should have already been printed
             //AC_LOG( DEBUG, "Picked profile: " << *rgb_profile );
 
-            AC_LOG(DEBUG, "Open...");
-            color_sensor->open({ rgb_profile });
-            AC_LOG(DEBUG, "Start...");
-
-            color_sensor->start(make_frame_callback([&](frame_holder fref) {}));
+            color_sensor->start_stream_for_calibration({ rgb_profile });
+           
+            if(false == color_sensor->is_streaming())
```

Comment:
This is not easily understood. Please explain.

Reply:
I expect that after the call to start_stream_for_calibration we are in one of this states:
* color stream opened and started by calibration process
* color stream is opened and started by user
*** color stream is opened by the user but not started yet.**

The last bullet has a LOG to it, it means AC did not start the sensor and still at this point don't have all the conditions it needs.

Is it clear now?

Reply:
So there's a corner case in which the user opened() but hasn't started(), therefore our start() cannot proceed and we have to wait?

Reply:
We are not really waiting... the start function will continue.
The color processing block will not be called so AC won't start...
if the user will leave the sensor at open state with no start, ac will not work.

But I think it is not the case.

On the viewer + on the examples we always guide to open and start continuously.

I don't think it worth special logic cause it's a complex situation...

What do you think?

Reply:
I'm more concerned with usage that's non-standard, like from some API that someone writes. For example, Miri has an interesting bug where she gets different results with open() vs open()+start().

What if we started() the stream ourselves, and transferred ownership, then? If the user then tried to start() again, it would close() and reopen. If we make the start() smart enough to recognize that the profile hasn't changed, maybe we can even make it seamless...? We can do it in a followup PR if you prefer.

If we leave it like this, the message should be a warning.
Please comment in the code all that we've written here.

Reply:
I am not sure I understand what you mean.
lets comment and merge and speak on another PR.